### ISSUE TITLE
C preprocessor counts number of args wrongly

### ISSUE THREAD

**Issue by amikstcyr** [libocca/occa#312]
A macro called "superclever" injects the argument list in a function called "callme".
Macro expands correctly with cpp (gcc) and Intel 2019 compilers. OCCA's c preprocessors seems to assume that comas determine the number of arguments.

In the provided codes, comment or uncomment the callme and superclever calls in addVectors.okl. superclever expands to how callme is written.

```
<SNIP>/raw_source.cpp:2:19: Included file:
comas.h:1:9: Expanded from macro 'COMBINE_':
<SNIP>/raw_source.cpp:10:3: Error: Too many arguments, expected 2 arguments
  superclever(ab);
  ^
```

To reproduce (very similar to #311 ):

addVectors.okl:
-----------------

```cpp
#include "comas.h"

#include "macros.h"

@kernel void addVectors(const int entries,
                        const float *a,
                        const float *b,
                        float *ab) {
  superclever(ab);
  //callme(&ab[0], &ab[1] ? &ab[1] : &ab[0] , &ab[2] ? &ab[2] : &ab[1], &ab[3] ? &ab[3] : &ab[2] );
  for (int i = 0; i < entries; ++i; @tile(16, @outer, @inner)) {
    ab[i] = a[i] + b[i];
  }
}
```

comas.h:
----------

```cpp
#define COMBINE_(x, y) x y
inline void callme(float *a, float *b, float *c, float *d){ *a;}
```

macros.h:
-----------

```cpp
#define superclever(z) COMBINE_(callme, (&z[0], &z[1] ? &z[1] : &z[0] , &z[2] ? &z[2] : &z[1], &z[3] ? &z[3] : &z[2] ))
```

**Comment by dmed256** [comment]
This seems to be due to the filter streams keeping the last seen token and not clearing it when rewinding. I'll have to add a method that will rewind things downstream.

**Comment by dmed256** [comment]
Fixed in #315
